dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/JdbcTrackedEntityChangeLogStore.java

```diff
@@ -106,28 +116,42 @@ public Page<TrackedEntityChangeLog> getTrackedEntityChangeLog(
                   join trackedentity t using(trackedentityid)
                   join trackedentityattribute tea using(trackedentityattributeid)
                   join userinfo u on u.username = audit.modifiedby
-                  where audit.audittype in ('CREATE', 'UPDATE', 'DELETE')
-                  and t.uid = :trackedEntity
        """;
-    MapSqlParameterSource parameters =
-        new MapSqlParameterSource("trackedEntity", trackedEntity.getValue())
-            .addValue("limit", pageParams.getPageSize() + 1)
-            .addValue("offset", (pageParams.getPage() - 1) * pageParams.getPageSize());
+
+    if (program != null) {
+      sql +=
+          """
+              join program_attributes pa using (trackedentityattributeid)
+              join program p using (programid)
+              where audit.audittype in ('CREATE', 'UPDATE', 'DELETE')
+              and t.uid = :trackedEntity
+              and p.uid = :programUid
+          """;
+      parameters.addValue("programUid", program.getValue());
+    } else {
+      sql +=
+          """
+              where audit.audittype in ('CREATE', 'UPDATE', 'DELETE')
+              and t.uid = :trackedEntity
+          """;
+    }
 
     List<TrackedEntityChangeLog> changeLogs;
     if (attributes.isEmpty()) {
       sql +=
           """
-              order by %s) cl
-              limit :limit offset :offset
-          """.formatted(sortExpressions(order));
+              order by %s
```

Comment:
It will in the future. I don't think it needs to handle something that is not supported yet. We would also need a test for multiple expressions if we implement it but we don't yet support that.